test/eventHandlers/WorkerInitHandler.test.ts

```diff
@@ -117,25 +117,125 @@ describe('WorkerInitHandler', () => {
         );
     });
 
-    it('Fails for missing entry point', async function (this: ITestCallbackContext) {
-        const fileSubpath = await setTestAppMainField('missing.js');
+    describe('entry point error', () => {
+        async function verifyInitSucceedsAndLogsError(): Promise<void> {
+            const fileSubpath = await setTestAppMainField('throwError.js');
+            stream.addTestMessage(msg.init.request(testAppPath));
+            const errorMessage = `Worker was unable to load entry point "${fileSubpath}": test`;
+            await stream.assertCalledWith(
+                msg.init.receivedRequestLog,
+                msg.loadingEntryPoint(fileSubpath),
+                msg.errorLog(errorMessage),
+                msg.init.response
+            );
+        }
 
-        stream.addTestMessage(msg.init.request(testAppPath));
-        const warningMessage = `Worker was unable to load entry point "${fileSubpath}": Found zero files matching the supplied pattern`;
-        await stream.assertCalledWith(msg.init.receivedRequestLog, msg.warningLog(warningMessage), msg.init.response);
-    });
+        async function verifyInitFails(): Promise<void> {
+            const fileSubpath = await setTestAppMainField('throwError.js');
+            stream.addTestMessage(msg.init.request(testAppPath));
+            const errorMessage = `Worker was unable to load entry point "${fileSubpath}": test`;
+            await stream.assertCalledWith(
+                msg.init.receivedRequestLog,
+                msg.loadingEntryPoint(fileSubpath),
+                msg.init.failedResponse(errorMessage)
+            );
+        }
 
-    it('Fails for invalid entry point', async function (this: ITestCallbackContext) {
-        const fileSubpath = await setTestAppMainField('throwError.js');
+        describe('Node >=v20', () => {
+            before(function (this: IHookCallbackContext) {
+                if (!isNode20Plus()) {
+                    this.skip();
+                }

```

Comment:
Because it can get really messy/confusing otherwise, I'd be in favor of adding a line here to clear environment variables, and set the desired app setting explicitly in each test case

Reply:
env vars get cleared after each test here: https://github.com/Azure/azure-functions-nodejs-worker/blob/v3.x/test/eventHandlers/TestEventStream.ts#L83